

syntax = "proto3";

package communication;

message MarkerRequest {
    string TYPE = 1;
    int64 M_ROUND = 2;
    string SENDER = 3;
    int64 NB = 4;
}

message MarkerReply {
    string TYPE = 1;

}

message MessageRequest{
    string TYPE = 1;
}
message MessageReply{
    string TYPE = 1;
}


service Remote {

    rpc Marker(MarkerRequest) returns (MarkerReply); 
    rpc Message(MessageRequest) returns (MessageReply);
}

//package name
//name of messages
//fields in messages

//generate the files with this
//python3 -m grpc_tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/communication.proto
//WITH THIS
//python3 -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. communication.proto